aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -531,6 +531,63 @@ void triangular_solve_kernel(Tensor& A, Tensor& B, Tensor& infos, bool upper, bo
   });
 }
 
+/*
+  Solves the matrix equation A X = B
+  X and B are n-by-nrhs matrices, A is represented using the LU factorization.
+  This is an in-place routine, content of `b` is overwritten.
+
+  Args:
+  * `b` -  [in] the right hand side matrix B
+           [out] the solution matrix X
+  * `lu` - [in] the LU factorization of matrix A (see at::_lu_with_info)
+  * `pivots` - [in] the pivot indices (see at::_lu_with_info)
+
+  For further details, please see the LAPACK documentation for GETRS.
+*/
+template <typename scalar_t>
+void apply_lu_solve(const Tensor& b, const Tensor& lu, const Tensor& pivots) {
+#ifndef USE_LAPACK
+  TORCH_CHECK(
+      false,
+      "Calling torch.lu_solve on a CPU tensor requires compiling ",
+      "PyTorch with LAPACK. Please use PyTorch built with LAPACK support.");
+#else
+  char trans = 'N';
+  auto b_data = b.data_ptr<scalar_t>();
+  auto lu_data = lu.data_ptr<scalar_t>();
+  auto pivots_data = pivots.data_ptr<int>();
+  auto b_stride = matrixStride(b);
+  auto lu_stride = matrixStride(lu);
+  auto pivots_stride = pivots.size(-1);
+  auto batch_size = batchCount(b);
+
+  auto n = lu.size(-2);
+  auto nrhs = b.size(-1);
+  auto leading_dimension = std::max<int64_t>(1, n);
+
+  int info = 0;
+  for (const auto i : c10::irange(batch_size)) {
+    scalar_t* b_working_ptr = &b_data[i * b_stride];
+    scalar_t* lu_working_ptr = &lu_data[i * lu_stride];
+    int* pivots_working_ptr = &pivots_data[i * pivots_stride];
```

Comment:
Could we use `batched_iterator_with broadcasting` here? This is to avoid memory inefficiencies and also simplify the code a bit...

Reply:
I was hesitant to use it because it requires further benchmarking. We are going to add cuSOLVER backend for this function soon which launches kernels asynchronously and I'm not sure how would `batched_iterator_with broadcasting` behave in this situation.
And the specialized batched variants require full materialization of batched inputs.

Reply:
Got it, but this is the CPU code, so we do not depend on cuSOLVER here I presume?